test/e2e/pod_schedule_error_test.go

```diff
@@ -70,16 +72,24 @@ func TestPodScheduleError(t *testing.T) {
 	}
 
 	names.Config = serviceresourcenames.Configuration(svc)
+	names.Route = serviceresourcenames.Route(svc)
 
 	err = v1a1test.WaitForServiceState(clients.ServingAlphaClient, names.Service, func(r *v1alpha1.Service) (bool, error) {
+		// TODO(taragu): after PR #4594, Route's status will be propagated to the Service level if
+		// there's no ready Revision, therefore we could receive either err message from Configuration or
+		// from Route. The error propagated from Route is not as informative to users comparing to the
+		// error propagated from Configuration, so we would like to improve the error coming from Route.
 		cond := r.Status.GetCondition(v1alpha1.ConfigurationConditionReady)
```

Comment:
This condition is looking at "ConfigurationsReady" which should never receive Ready information from the Route. It should only contain the propagated up information from the Configuration

Reply:
@dgerd Thank you so much for helping me figure this out. The test was actually check for the wrong condition. It was checking `ConfigurationConditionReady` (which is `apis.ConditionReady`) [here](https://github.com/knative/serving/blob/master/test/e2e/pod_schedule_error_test.go#L75) when trying to check the Service's Configuration status. It should be `ServiceConditionConfigurationsReady`.